flink-core/src/main/java/org/apache/flink/api/common/state/ListState.java

```diff
@@ -48,10 +48,13 @@ public interface ListState<T> extends MergingState<T, Iterable<T>> {
      * given list of values. The next time {@link #get()} is called (for the same state partition)
      * the returned state will represent the updated list.
      *
-     * <p>If null or an empty list is passed in, the state value will be null.
+     * <p>If an empty list is passed in, the state value will be null.
```

Comment:
Maybe it's also better to unify the behavious of null and empty list in the future.
But this may modify the semantic of public API so it could be done in 2.0. WDYT?

Reply:
Agreed. A more unified behaviour makes sense. Let's do this in 2.0.